src/feature/hs/hs_descriptor.c

```diff
@@ -2277,12 +2274,14 @@ desc_decode_encrypted_v3(const hs_descriptor_t *desc,
        * authorization is failing. */
       log_warn(LD_REND, "Client authorization for requested onion address "
                         "is invalid. Can't decrypt the descriptor.");
+      ret = HS_DESC_DECODE_BAD_CLIENT_AUTH;
     } else {
       /* Inform at notice level that the onion address requested can't be
        * reached without client authorization most likely. */
       log_notice(LD_REND, "Fail to decrypt descriptor for requested onion "
                         "address. It is likely requiring client "
                         "authorization.");
+      ret = HS_DESC_DECODE_NEED_CLIENT_AUTH;
```

Comment:
Blocker: This does not seem to work smoothly for me. In particular, it takes 2 minutes before the SOCKS error gets propagated to my torsocks. This can't work for a browser tab, since the user will have long closed it.
```
         Jun 12 13:21:28.352 [notice] Fail to decrypt descriptor for requested onion address. It is likely requiring client authorization.
         Jun 12 13:21:38.389 [notice] Fail to decrypt descriptor for requested onion address. It is likely requiring client authorization.
         Jun 12 13:21:55.468 [notice] Fail to decrypt descriptor for requested onion address. It is likely requiring client authorization.
         Jun 12 13:22:16.417 [notice] Fail to decrypt descriptor for requested onion address. It is likely requiring client authorization.
         Jun 12 13:22:37.447 [notice] Fail to decrypt descriptor for requested onion address. It is likely requiring client authorization.
         Jun 12 13:22:55.453 [notice] Fail to decrypt descriptor for requested onion address. It is likely requiring client authorization.
         Jun 12 13:23:17.476 [notice] Fail to decrypt descriptor for requested onion address. It is likely requiring client authorization.
         Jun 12 13:23:28.315 [notice] Tried for 120 seconds to get a connection to 6eqv54nzmho6tmgbwfm76cciveryhunmz62c3tm4ez6wsbzricz5ysad:5858. Giving up. (waiting for rendezvous desc)
         ...
         1560335008 ERROR torsocks[10662]: Socks5 server replied an unknown code 244 (in socks5_recv_connect_reply() at socks5.c:559)
```

Reply:
Yeah, this is the SOCKS timeout that mcs/brade reported... It needs to be investigated separately for now. Ticket is about to be opened about this.